text/0000-benchmarking.md

```diff
@@ -129,8 +146,28 @@ Samples are [winsorized], so extreme outliers get clamped.
 `cargo bench` essentially takes the same flags as `cargo test`, except it has a `--bench foo`
 flag to select a single benchmark target.
 
+[winsorized]: https://en.wikipedia.org/wiki/Winsorizing
+
+## `mem::black_box`
+
+```rust
+pub fn black_box<T>(x: T) -> T;
+```
+
+Prevents a value or the result of an expression from being optimized away by the
+compiler adding as little overhead as possible. It does not prevent
```

Comment:
Also mention that it prevents outer code from "seeing" the inside optimization-wise -- values produced by black_box are from the optimizer's perspective no different from being random.